### DISCUSSION TITLE
RSS feed generation support

### DISCUSSION THREAD

**Discussion by lggomez** [zim-desktop-wiki/zim-desktop-wiki#1816] [Ideas]
I created a program which scaffolds an RSS feed given the root directory of a notebook and some parameters: https://github.com/lggomez/go-zimrss

It currently works as is, in a separate way from the site publication and must be done manually. Since python is not my first language and I'd want to validate the reusability value of this from the main application without having to reimplement this, I have some questions for the maintainers/collaborators:

1. What is your opinion on having such a feature? (optional RSS feed generation for the wiki pages on HTML export)
2. How do you think would be the best approach to integrate this feature? should it warrant changes in the core app or should a plugin suffice?
3. Given that this program is a third party dependency, how is your preferred way of handling it? go makes it easy to install and build applications once you install the main package it but I also understand if your preferred flavor is a debian/ubuntu package of go-zimrss itself

**Comment by introt** [comment]
Hi!

Sorry for the delay, I seem to have missed this even though I was the one who converted the ticket into a discussion - though back then I'd certainly have had even less of an idea of how things work around here.

1. I'd imagine RSS feed generation would be welcomed by those who use Zim as a CMS. Feeds certainly seem popular in the public personal wiki space, though I personally haven't seen much Zim usage there.
2. Since it's an external program that operates on the wiki source files themselves, I'd go for a [custom build script](https://github.com/jaap-karssenberg/zim-wiki/wiki/Build-a-Website). I'd imagine most users have one tailored to their workflow already, including commands to eg. push the site to the server using git/ftp/etc. This can then be triggered via a Custom Tool in the UI. For better integration, implementing #2000 could certainly help with the missing metadata.
3. As far as I know, our current way of handling external third party stuff is having a [public wiki](https://github.com/jaap-karssenberg/zim-wiki/wiki/) that anyone can add them to. Feel free to distribute your program in any way you like.
